sdk/communication/azure-communication-common/src/main/java/com/azure/android/communication/common/CommunicationCloudEnvironment.java

```diff
@@ -7,36 +7,45 @@
 /**
  * The cloud that the identifier belongs to.
  */
-public final class CommunicationCloudEnvironment{
+public final class CommunicationCloudEnvironment {
     private static final String PUBLIC_VALUE = "public";
     private static final String DOD_VALUE = "dod";
     private static final String GCCH_VALUE = "gcch";
 
-    private final String environmentValue;
+    private String environmentValue;
 
     /**
-     * Create CommunicationCloudEnvironment with name string
-     * @param environmentValue name of hte cloud environment
+     * Creates a default CommunicationCloudEnvironment
      */
-    public CommunicationCloudEnvironment(String environmentValue) {
+    public CommunicationCloudEnvironment() {
+        this.environmentValue = PUBLIC_VALUE;
+    }
+
+    /**
+     * Sets an environment value from a String
+     * @param environmentValue string representation of the environment value
+     * @return CommunicationCloudEnvironment object itself
+     */
+    public final CommunicationCloudEnvironment fromString(String environmentValue) {
```

Comment:
We have a feedback on the Java ApiView to make this `static`.

Reply:
@RezaJooyandeh  If we make this static, then we cannot refer to "this" in the function implementation. Then should we be changing environmentValue to static as well?

Reply:
No, essentially this would be a static method that instantiate a new `CommunicationCloudEnvironment` based in the input and returns it.